pkg/proto/nbcontract/v1/kubeletconfig.proto

```diff
@@ -1,15 +1,18 @@
 syntax = "proto3";
 package nbcontract.v1;
+import "pkg/proto/nbcontract/v1/featurestate.proto";
 
-message KubeletConfiguration {
+message KubeletConfig {
   // TODO(ace): remove these/make api defensible
-  repeated string kubelet_flags = 1;
+  map<string, string> kubelet_flags = 1;
```

Comment:
Not disagreeing, but flagging this as an impactful design decision. Order might actually matter here, need to review kubelet flags to make a final call. (Also need a decision on supporting generic flags vs. breaking them out into individual fields.)

Reply:
Okay I didn't realize it's in alphabetical order. I will check with the team to understand why it's needed.
Assuming it's needed, then I think you are right, I should change it back to a list of string.
What I thought was that as the format of kubelet_flags are key-value pairs, using string map should be more suitable.

Reply:
Alphabetical order? When kubelet gets these as command line arguments, the order would matter - though probalby only if there are duplicates, which would not be the case with map. Modeling this is as map is probably fine - especially considering KubeletConfig exists, and does essentially the same. Alternatively, we could consider breaking them out - though evolving this would be challenging.

Reply:
In this line, it seems it's doing an alphabetical ordering to the keys. https://github.com/Azure/AgentBaker/blob/df3496f1dac695cc13693096cb6fa6ac3a88a4c5/pkg/agent/utils.go#L373
I didn't compile it to see the results though.
As for breaking them out, I think that will be considered in our next step.

Reply:
Ok, I think we can take as yet another datapoint showing (perhaps counterintuitively?) that order does not actually matter. For actual sorting, I suspect AgentBaker does it just to make the resulting CSE deterministic, maybe even just for testability?

Reply:
Could be. Let me check with the team to see if anyone knows.

Reply:
Team thinks it could be just for human readability to know what flags are present. Let's keep it as is and we will review this in this milestone.

Reply:
Sure. 

(Surprised testability is not the reason. Doesn't AB have tests where it compares what gets generated with expected output - including CustomData, like [this one](https://github.com/Azure/AgentBaker/blob/9d9d6a659c7af00c5d8bc6816d66634ac7a58410/pkg/agent/testdata/AKSUbuntu2204%2BSecureTLSBoostrapping/CustomData#L263)? Comparing the results won't work unless the order in the output is deterministic - the same every time - and sorting is a way to enforce it ...)

Reply:
Your point could be another reason. I didn't (know how to) run the test yet though.